aggregates/account/src/main/java/com/bloxbean/cardano/yaci/store/account/storage/impl/AccountBalanceStorageImpl.java

```diff
@@ -80,18 +84,25 @@ public List<AddressBalance> getAddressBalance(String address) {
     @Transactional
     @Override
     public void saveAddressBalances(@NonNull List<AddressBalance> addressBalances) {
-        List<AddressBalanceEntity> entities = addressBalances.stream().map(mapper::toAddressBalanceEntity)
+        List<AddressBalanceEntity> addressBalanceEntities = addressBalances.stream().map(mapper::toAddressBalanceEntity)
                 .toList();
+        List<AddressEntity> addressEntities = addressBalances.stream().map(mapper::toAddressEntity)
+                .collect(Collectors.toMap(AddressEntity::getAddress, Function.identity(), (a,b) -> a))
+                .values()
```

Comment:
Or, should we just collect it in a Set for distinct values?

Reply:
Can we just pass List of AddressBalance to saveAddresses method and handle it there?
As we will be using only JOOQ for address saving, we don't need to convert it to AddressEntity.

Something like this I had done before to track address
https://github.com/bloxbean/yaci-store/blob/41c4383a1db6c9bc4b87f21391c029e06fcdf266/aggregates/account/src/main/java/com/bloxbean/cardano/yaci/store/account/storage/impl/AccountBalanceStorageImpl.java#L159

Reply:
Fyi. Now there is a AddressUtil class in account module which can give address and full address for an address. So we can use it to populate full address.